enterprise_access/apps/subsidy_access_policy/models.py

```diff
@@ -0,0 +1,192 @@
+        if self.group_uuid in group_uuids:
+            if subsidy_client.get_license_for_group(self.subsidy_uuid, self.group_uuid):
+                return super().can_redeem(learner_id, content_key)
+
+        return False
+
+
+class LearnerCreditAccessPolicy(SubsidyAccessPolicy):
+    """
+    A subsidy access policy for learner credit.
+    
+    .. no_pii:
+    """
+    class Meta:
+        """
+        Metaclass for LearnerCreditAccessPolicy.
+        """
+        abstract = True
+
+
+class LicenseRequestAccessPolicy(SubscriptionAccessPolicy):
+    """
+    A subsidy access policy for subscriptions with a request type access method.
+    """
+    access_method = AccessMethods.REQUEST
+
+
+class LicenseAccessPolicy(SubscriptionAccessPolicy):
+    """
+    A subsidy access policy for subscriptions with a direct type access method.
+    """
+    access_method = AccessMethods.DIRECT
+
+
+class PerLearnerEnrollmentCapLearnerCreditAccessPolicy(LearnerCreditAccessPolicy):
+    """
+    Policy that limits the number of enrollments transactions for a learner in a subsidy.
+    """
+    per_learner_enrollment_limit = models.IntegerField(
+        blank=True,
+        default=0,
+    )
+
+    def can_redeem(self, learner_id, content_key):
+        learner_transaction_count = subsidy_client.transactions_for_learner(
+            subsidy_uuid = self.subsidy_uuid,
+            learner_id = learner_id,
+            ).count()
+        if learner_transaction_count < self.per_learner_enrollment_limit:
+            return super().can_redeem(learner_id, content_key)
+        
+        return False
+
+class PerLearnerSpendCapLearnerCreditAccessPolicy(LearnerCreditAccessPolicy):
+    """
+    Policy that limits the amount of learner spend for enrollment transactions.
+    """
+    per_learner_spend_limit = models.IntegerField(
+        blank=True,
+        default=0,
+    )
+
+    def can_redeem(self, learner_id, content_key):
+        spent_amount = subsidy_client.amount_spent_for_learner(
+            subsidy_uuid = self.subsidy_uuid,
+            learner_uuid = learner_id,
+            )
+        #TODO the following should also incorporate the course price
+        if spent_amount < self.per_learner_spend_limit:
+            return super().can_redeem(learner_id, content_key)
+        
+        return False
+
+class CappedEnrollmentLearnerCreditAccessPolicy(LearnerCreditAccessPolicy):
+    """
+    Policy that limits the maximum amount that can be spent aggregated across all users covered by this policy
+    """
+    spend_limit = models.IntegerField(
+        blank=True,
+        default=0,
+    )
+
+    def can_redeem(self, learner_id, content_key):
+            group_amount_spent = subsidy_client.amount_spent_for_group_and_catalog(
+                subsidy_uuid = self.subsidy_uuid,
+                group_uuid = self.group_uuid,
+                catalog_uuid = self.catalog_uuid,
+                )
+            if  group_amount_spent < self.spend_limit:
```

Comment:
@iloveagent57 Course price here as well ... right?

Reply:
Yup, the amount previously spent here is the aggregate of all transaction amounts for the "group" (which in v1 of this project, is just the whole enterprise).